app/email/email_batching.go

```diff
@@ -100,6 +100,17 @@ func (job *EmailBatchingJob) Start() {
 	}
 }
 
+// Stop will cancel the task properly, flushing out any pending notifications.
+// Although this still won't send those notifications which are yet to be sent
+// due to a user's PreferenceNameEmailInterval.
+func (job *EmailBatchingJob) Stop() {
+	job.taskMutex.Lock()
+	if task := job.task; task != nil {
+		task.Cancel()
+	}
+	job.taskMutex.Unlock()
```

Comment:
Is `task.Cancel()` a blocking function? If so, is it safe/expected to keep the lock for the whole time?

Reply:
Yes, that should be fine since this is only called while stopping the server.